processors/osf/processor.py

```diff
@@ -0,0 +1,146 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import json
+import logging
+import requests
+import datetime
+from .. import base
+logger = logging.getLogger(__name__)
+
+
+# Module API
+
+def process(conf, conn):
+
+    # Log started
+    logger.info('Started trials export')
+
+    # Export trials
+    # http://jamdb.readthedocs.io/en/latest/index.html
+    count = 0
+    limit = 100
+    token_ttl_seconds = 30*60  # api has limit 60*60 seconds
+    token_issued_time = datetime.datetime.now()
+    session = requests.Session()
+    for trials in _read_trial_groups(conn, limit=limit):
+
+        # Ensure authentificated
+        if (datetime.datetime.now() - token_issued_time).seconds > token_ttl_seconds:
+            del session.headers['Authorization']
+        if 'Authorization' not in session.headers:
+            url = '%s/auth' % conf['OSF_URL']
+            res = session.post(url, json={
+              'data': {
+                'type': 'users',
+                'attributes': {
+                  'provider': 'osf',
+                  'access_token': conf['OSF_KEY'],
+                }
+              }
+            })
+            # Check status
+            # 200 - ok
+            if res.status_code not in [200]:
```

Comment:
Normal flow here could be with many status codes so I use list by default to simplify changes.